lib/connect.js

```diff
@@ -633,7 +658,34 @@ function validateBandwidthProfile(bandwidthProfile) {
     }
     return typeof video[prop] === typeOrValues ? null
       : Error(`options.bandwidthProfile.video.${prop}`, typeOrValues);
+  }, validateRenderDimensions(video.renderDimensions));
+}
+
+function validateDimensions(prop, dimensions) {
+  if (typeof dimensions === 'undefined') {
+    return null;
+  }
+  if (dimensions === null || !isNonArrayObject(dimensions)) {
+    return E.INVALID_TYPE(`options.bandwidthProfile.video.renderDimensions.${prop}`, 'object');
+  }
+  return ['height', 'width'].reduce((error, dimension) => {
```

Comment:
I think using reduce for two properties is overkill. Would be easier to read w/o reduce.

```
if (typeof dimension.height !== `number`) {
   return E.INVALID_TYPE(`options.bandwidthProfile.video.renderDimensions.${prop}.height`, 'number');
}
if (typeof dimension.width !== `number`) {
   return E.INVALID_TYPE(`options.bandwidthProfile.video.renderDimensions.${prop}.width`, 'number');
}
```